department-of-veterans-affairs/abd-vro
# Add Dead Letter Queue and update existing queue configurations to use it

## What was the problem?
Messages that fail to process are lost.

Associated tickets or Slack threads:
- https://app.zenhub.com/workspaces/vro-team-6557e67173391c000e1409f3/issues/gh/department-of-veterans-affairs/abd-vro/2758

## How does this fix it?[^1]
When bip-api starts up and creates/configures queues, it will now also create a dead-letter queue. Messages that fail to process will be placed on this queue

## How to test this PR
- Build and launch the bip-api service in Docker
- Go to the RabbitMQ console and verify that the dead-letter queue was created
- Place some junk data on any of the queues. bip-api will fail to process it and you can see the message end up in the dead-letter queue.
- after implementing it, we discovered that coordinating changes for EPMerge and BIP are needed
- When we deploy this it will need to be planned to ensure we do not take down the EPMerge and BIP


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
